File: backend/app/schemas/analytics.py
from datetime import date
from decimal import Decimal
from typing import List, Optional

from pydantic import BaseModel


class SalesOverviewResponse(BaseModel):
    total_revenue: Decimal
    paid_orders: int
    active_rides: int
    new_rides: int
    average_order_amount: Decimal


class SalesTrendPoint(BaseModel):
    date: date
    orders: int
    revenue: Decimal
    average_order_amount: Decimal


class PriceTrendPoint(BaseModel):
    date: date
    product: str
    ride_count: int
    average_price_per_month: Decimal
    average_contact_price: Decimal
    minimum_price_per_month: Decimal
    maximum_price_per_month: Decimal


class ProductRankingItem(BaseModel):
    product: str
    product_label: str
    orders: int
    revenue: Decimal
    average_order_amount: Decimal


class RideRankingItem(BaseModel):
    ride_id: int
    ride_title: str
    product: str
    orders: int
    revenue: Decimal
    total_seats: int
    recruit_seats: int
    remaining_seats: int
    status: str
    owner_id: int
    owner_nickname: Optional[str] = None


class GmvByProduct(BaseModel):
    product: str
    product_label: str
    exercised_gmv: Decimal
    potential_gmv: Decimal
    remaining_gmv: Decimal
    exercised_count: int
    potential_count: int
    ride_count: int


class GmvResponse(BaseModel):
    exercised_gmv: Decimal
    potential_gmv: Decimal
    remaining_gmv: Decimal
    benchmark_gmv: Decimal
    exercised_count: int
    potential_count: int
    active_rides: int
    by_product: List[GmvByProduct]

File: frontend/src/api/analytics.ts
import apiClient from './client'

export interface SalesOverview {
  total_revenue: number
  paid_orders: number
  active_rides: number
  new_rides: number
  average_order_amount: number
}

export interface PlatformStats {
  total_users: number
  active_rides: number
  total_rides: number
}

export interface PriceTrendPoint {
  date: string
  product: string
  ride_count: number
  average_price_per_month: number
  average_contact_price: number
  minimum_price_per_month: number
  maximum_price_per_month: number
}

export interface RideRankingItem {
  ride_id: number
  ride_title: string
  product: string
  orders: number
  revenue: number
  total_seats: number
  recruit_seats: number
  remaining_seats: number
  status: string
  owner_id: number
  owner_nickname?: string
}

export interface ProductRankingItem {
  product: string
  product_label: string
  orders: number
  revenue: number
  average_order_amount: number
}

export interface GmvByProduct {
  product: string
  product_label: string
  exercised_gmv: number
  potential_gmv: number
  remaining_gmv: number
  exercised_count: number
  potential_count: number
  ride_count: number
}

export interface GmvResponse {
  exercised_gmv: number
  potential_gmv: number
  remaining_gmv: number
  benchmark_gmv: number
  exercised_count: number
  potential_count: number
  active_rides: number
  by_product: GmvByProduct[]
}

export const analyticsApi = {
  getPlatformStats() {
    return apiClient.get<PlatformStats>('/analytics/platform/stats')
  },

  getSalesOverview(days = 30) {
    return apiClient.get<SalesOverview>('/analytics/sales/overview', { params: { days } })
  },

  getPriceTrends(days = 180) {
    return apiClient.get<PriceTrendPoint[]>('/analytics/prices/trends', { params: { days } })
  },

  getRideRankings(days = 30, limit = 5) {
    return apiClient.get<RideRankingItem[]>('/analytics/rankings/rides', { params: { days, limit } })
  },

  getProductRankings(days = 30, limit = 10) {
    return apiClient.get<ProductRankingItem[]>('/analytics/rankings/products', { params: { days, limit } })
  },

  getGmv() {
    return apiClient.get<GmvResponse>('/analytics/gmv')
  },
}
